docs/debug-os-level.md

```diff
@@ -0,0 +1,74 @@
+# Debugging NIO programs from the OS level
+
+
+## List all file descriptors
+
+```
+lsof -p $(pgrep NIOEchoServer)
```

Comment:
Is it worth noting that you should use the process name of the process you're trying to debug here? Won't necessarily be obvious to everyone that `NIOEchoServer` isn't a weird magic string.